pubtools/pulplib/_impl/model/repository/file.py

```diff
@@ -20,3 +29,66 @@ class FileRepository(Repository):
     )
 
     mutable_urls = attr.ib(default=attr.Factory(lambda: ["PULP_MANIFEST"]), type=list)
+
+    def upload_file(self, file_obj, relative_url=None):
+        """Upload a file to this repository.
+
+        Args:
+            file_obj (str, file object)
+                If it's a string, then it's the path of file to upload.
+                Else, it ought to be a file-like object
+
+            relative_url (str)
+                Path that should be used in remote repository
+
+                If omitted, filename will be used.
+        Returns:
+            Future[:class:`~pubtools.pulplib.Task`]
+                A future which is resolved when import succeeds.
+
+                The future contains the task to import uploaded content
+                to repository
+
+        Raises:
+            DetachedException
+                If this instance is not attached to a Pulp client.
```

Comment:
Could you please use sphinx "versionadded" directive here to document when we've added this new API?

e.g.

```
  .. versionadded:: 1.2.0
```